build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmExtractorProducer.java

```diff
@@ -0,0 +1,50 @@
+package org.jfrog.build.extractor.npm.extractor;
+
+import org.apache.commons.lang.StringUtils;
+import org.jfrog.build.extractor.npm.types.NpmPackageInfo;
+import org.jfrog.build.extractor.producerConsumer.ProducerRunnableBase;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import java.util.Enumeration;
+
+/**
+ * Produces PackageInfos from the dependencies tree.
```

Comment:
I suggest elaborating more here about how the whole process works, because it is not trivial.